src/diagnostics/OpenPMDWriter.H

```diff
@@ -82,24 +83,25 @@ private:
      * \param[in] varnames list of variable names for the fields (ExmBy, EypBx, Ey, ...)
      * \param[in,out] iteration openPMD iteration to which the data is written
      * \param[in] output_step current time step to dump
+     * \param[in] lev MR level
      */
     void WriteFieldData (amrex::FArrayBox const& fab, amrex::Geometry const& geom,
                          const int slice_dir, const amrex::Vector< std::string > varnames,
-                         openPMD::Iteration iteration, const int output_step);
+                         openPMD::Iteration iteration, const int output_step, const int lev);
 
     /** Named Beam SoA attributes per particle as defined in BeamIdx
      */
     amrex::Vector<std::string> m_real_names {"weighting", "momentum_x", "momentum_y", "momentum_z"};
 
-    /** Parallel openPMD-api Series object for output */
-    std::unique_ptr< openPMD::Series > m_outputSeries;
+    /** vector over levels of openPMD-api Series object for output */
+    amrex::Vector<std::unique_ptr< openPMD::Series >> m_outputSeries;
 
     /** openPMD backend: h5, bp, or json. Default depends on what is available */
     std::string m_openpmd_backend = "default";
 
     /** Last iteration that was written to file.
      * This is stored to make sure we don't write the last iteration multiple times. */
-    int m_last_output_dumped = -1;
+    amrex::Vector<int> m_last_output_dumped;
```

Comment:
Is this really a per-level quantity? In other words, is there a chance that we dump the last output for one level and not for the other? If not, this could just be 1 number, not a vector over levels.

Reply:
This is indeed a per-level quantity. Maybe the naming could be improved. This quantity is used to flag whether the openPMD dataset needs to be reset, which should only be done once per file (and not when we write to the same file again). So if this is not done per level, the data set of the next level will not be reset properly and it will crash. I forgot about this initially, that's how I know.

Reply:
Alright, that's fine then, thanks!